### ISSUE TITLE
range(1,1) returns empty array

### ISSUE THREAD

**Issue by drauschenbach** [Yonaba/Moses#38]
I'm pretty sure that `range(1,1)` should return `{1}`.

**Comment by drauschenbach** [comment]
```lua
> _.print(moses.range(1, 0))
{}
> _.print(moses.range(1, 1))
{}
> _.print(moses.range(1, 2))
{1, 2}
> _.print(moses.range(1, 3))
{1, 2, 3}
```

**Comment by chessai** [comment]
tl;dr: I agree, if a range is supposed to be isomorphic to the mathematical object of an interval, then range(1,1) is isomorphic to [1,1], which is just the degenerate interval.

This function works as according to the API docs:

"[_.range] --- Produces a flexible list of numbers. If one positive value is passed, will count from 0 to that value, with a default step of 1. If two values are passed, will count from the first one to the second one, with the same default step of 1. A third value passed will be considered a step value."

The problem is the inconsistency w.r.t. clusivity of endpoints, exactly as your examples point out.

_range(1,1) = {}            (1)
_range(1,2) = {1,2}       (2)
_range(1,3) = {1,2,3}    (3)

In (2) and (3), the endpoints are treated as exclusive, while in (1), they are not. This is because it counts from 1 to 1, with a default step of 1.

**Comment by drauschenbach** [comment]
I use the following patched range function in my Spark runtime lib [Stuart](https://github.com/BixData/stuart/blob/0.1.1/src/stuart/mosesPatchedRange.lua).

My patch is necessary because porting Spark involves porting Scala's semantics for a range function, which happens to match the lodash/underscore semantics. And that is really my benchmark for why moses is currently the anomaly on its unusable treatment of (1,1) params.

**Comment by Yonaba** [comment]
Thanks @drauschenbach and @chessai for your suggestions. It was implemented via commit 7c01770.
